src/Segment/GetAllSegments.php

```diff
@@ -1,7 +1,42 @@
 <?php
 namespace Clearbooks\Labs\Segment;
 
-class GetAllSegments 
+use Clearbooks\Labs\Segment\Gateway\SegmentProvider;
+use Clearbooks\Labs\Segment\GetAllSegments\ResponseSegment;
+use Clearbooks\Labs\Segment\UseCase\GetAllSegments\Response;
+
+class GetAllSegments implements Response
 {
+    /**
+     * @var SegmentProvider
+     */
+    private $segmentProvider;
+
+    /**
+     * @param SegmentProvider $segmentProvider
+     */
+    public function __construct( SegmentProvider $segmentProvider )
+    {
+        $this->segmentProvider = $segmentProvider;
+    }
+
+    /**
+     * @return Response
+     */
+    public function execute()
+    {
+        return $this;
```

Comment:
Isn't this the case of misplaced responsibilities? You are NOT executing anything!
Uncle Bob third test in Bowling Game Kata:
public class Game {
  private int score = 0;

  public void roll(int pins) {
    score += pins;
  }

  public int score() {
    return score;
  }
}